growattServer/__init__.py

```diff
@@ -128,106 +132,439 @@ def login(self, username, password, is_password_hashed=False):
+
+        Raises:
+            Exception: If the request to the server fails.
         """
-        response = self.session.get(self.get_url('newTlxApi.do'), params={
+        response = self.session.get(self.__get_url('newTlxApi.do'), params={
             'op': 'getTlxDetailData',
             'id': tlx_id
         })
 
-        data = json.loads(response.content.decode('utf-8'))
-        return data
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise Exception('Failed to retrieve detailed TLX inverter data')
+
+    def tlx_params(self, tlx_id):
+        """
+        Get parameters for TLX inverter.
+
+        Args:
+            tlx_id (str): The ID of the TLX inverter.
+
+        Returns:
+            dict: A dictionary containing the TLX inverter parameters.
+
+        Raises:
+            Exception: If the request to the server fails.
+        """
+        response = self.session.get(self.__get_url('newTlxApi.do'), params={
+            'op': 'getTlxParams',
+            'id': tlx_id 
+        })
+
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise Exception('Failed to retrieve TLX inverter parameters')
+
+    def tlx_get_all_settings(self, tlx_id):
+        """
+        Get all possible settings from TLX inverter.
+
+        Args:
+            tlx_id (str): The ID of the TLX inverter.
+
+        Returns:
+            dict: A dictionary containing all possible settings for the TLX inverter.
+
+        Raises:
+            Exception: If the request to the server fails.
+        """
+        response = self.session.post(self.__get_url('newTlxApi.do'), params={
+            'op': 'getTlxSetData'
+        }, data={
+            'serialNum': tlx_id
+        })
+
+        if response.status_code == 200:
+            data = response.json()
+            return data['obj']['tlxSetBean']
+        else:
+            raise Exception('Failed to retrieve TLX inverter settings')
+
+    def tlx_get_enabled_settings(self, tlx_id):
+        """
+        Get "Enabled settings" from TLX inverter.
+        
+        Args:
+            tlx_id (str): The ID of the TLX inverter.
+        
+        Returns:
+            dict: A dictionary containing the enabled settings.
+        
+        Raises:
+            Exception: If the request to the server fails.
+        """
+        string_time = datetime.datetime.now().strftime('%Y-%m-%d')
+        response = self.session.post(
+            self.__get_url('newLoginAPI.do'),
+            params={'op': 'getSetPass'},
+            data={'deviceSn': tlx_id, 'stringTime': string_time, 'type': '5'}
+        )
+
+        if response.status_code == 200:
+            return response.json().get('obj', {})
+        else:
+            raise Exception('Failed to retrieve enabled inverter settings')
+
+    def get_battery_info(self, serial_num):
```

Comment:
```suggestion
    def tlx_battery_info(self, serial_num):
```

In case other systems show up with batteries